x-pack/plugins/inference/public/util/create_observable_from_http_response.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { createParser } from 'eventsource-parser';
+import { HttpResponse } from '@kbn/core/public';
+import { Observable, throwError } from 'rxjs';
+import { createInferenceInternalError } from '../../common/errors';
+
+export function createObservableFromHttpResponse(response: HttpResponse): Observable<string> {
+  const rawResponse = response.response;
+
+  const body = rawResponse?.body;
+  if (!body) {
+    return throwError(() => {
+      throw createInferenceInternalError(`No readable stream found in response`);
+    });
+  }
+
+  return new Observable<string>((subscriber) => {
```

Comment:
IIRC, we are using an older version of `eventsource-parser` because of module interop issues, and it doesn't have EventSourceParserStream, but I can look into it again.

Reply:
I had a look, but running into some issues with CommonJS and availability of TextDecoderStream in the browser, so I'll punt on this one.